app/controllers/registrations_controller.rb

```diff
@@ -699,22 +699,25 @@ def refund_payment
     registration = charge.root_record.payment_intent.holder
     uses_v2 = registration.is_a? MicroserviceRegistration
 
+    redirect_path = uses_v2 ? edit_registration_v2_path(competition_id, registration.user_id) : edit_registration_path(registration)
+
     unless stripe_integration.present?
       flash[:danger] = "You cannot emit refund for this competition anymore. Please use your Stripe dashboard to do so."
-      return redirect_to edit_registration_path(registration)
+      return redirect_to redirect_path
     end
 
     refund_amount_param = params.require(:payment).require(:refund_amount)
     refund_amount = refund_amount_param.to_i
+    amount_left = charge.ruby_amount_available_for_refund - refund_amount
 
-    if refund_amount > charge.ruby_amount_available_for_refund
+    if amount_left < 0
```

Comment:
Methinks Ruby has `some_integer.negative?` which might be a tad bit more readable.